gobblin-service/src/test/java/org/apache/gobblin/service/modules/core/GobblinServiceHATest.java

```diff
@@ -255,6 +251,8 @@ public void cleanUp() throws Exception {
     cleanUpDir(COMMON_SPEC_STORE_PARENT_DIR);
 
     mysql.stop();
+    // `.close()` to avoid (in the aggregate, across multiple suites) - java.sql.SQLNonTransientConnectionException: Too many connections
+    testMetastoreDatabase.close();
```

Comment:
what if line 253 throws the exception and this statement is never reached

Reply:
true, but the canonical rule is to close in reverse order of opening.  because, this was the first one to be opened, it should be the last closed

Reply:
aah, I misunderstood the gist of your Q earlier and now see what you were getting at.  just made the change